### ISSUE TITLE
Crash in Observation+Combine l.36

### ISSUE THREAD

**Issue by benrudhart** [sindresorhus/Defaults#38]
Hi there,
I'm using a combine publisher like this
```swift
    private var myPublisher: NonFailingPublisher<Foo?> {
        Defaults
            .publisher(<FooName>, options: [.initial, .new])
            .compactMap { $0.newValue }
            .eraseToAnyPublisher()
    }
```
the output is assigned (through some mapping) to a `text` in a `UILabel`.

But when executed Defaults crashes in Observation+Combine l.36:
```swift
private func observationCallback(_ change: BaseChange) {
	_ = subscriber?.receive(change)
}
```
reason: `Thread 1: Fatal error: Invalid state: Received value before receiving subscription`

Maybe this issue is helpful (same error reason): https://github.com/CombineCommunity/RxCombine/pull/6

One (hacky) way to fix this could be to dispatch `_ = subscriber?.receive(change)`  to the next run loop via `DispatchQueue.main.async { ... }`

**Comment by sindresorhus** [comment]
// @fredyshox

**Comment by fredyshox** [comment]
I can look into that.

**Comment by benrudhart** [comment]
thank you!

**Comment by fredyshox** [comment]
I fixed this bug in #39, sorry it took so long. It's weird, because I've been personally using this API in various projects, but never with `.initial` option, which caused issue here.

**Comment by benrudhart** [comment]
Thank you for fixing this! 👏
